Tribler/Test/Core/test_downloadstate.py

```diff
@@ -0,0 +1,109 @@
+from Tribler.Core.DownloadState import DownloadState
+from Tribler.Core.simpledefs import DLSTATUS_DOWNLOADING, UPLOAD, DOWNLOAD, DLSTATUS_STOPPED
+from Tribler.Test.Core.base_test import TriblerCoreTest, MockObject
+
+
+class TestDownloadState(TriblerCoreTest):
+    """
+    This class contains tests for the download state.
+    """
+
+    def setUp(self, annotate=True):
+        TriblerCoreTest.setUp(self, annotate=annotate)
+        self.mock_download = MockObject()
+        mocked_tdef = MockObject()
+        mocked_tdef.get_name = lambda: "test"
+        mocked_tdef.get_length = lambda _: 43
+        self.mock_download.get_def = lambda: mocked_tdef
+
+    def test_getters_setters(self):
+        """
+        Testing various getters and setters in DownloadState
+        """
+        download_state = DownloadState(self.mock_download, DLSTATUS_DOWNLOADING, None, 0.5)
+        self.assertEqual(download_state.get_download(), self.mock_download)
+        self.assertEqual(download_state.get_progress(), 0.5)
+        self.assertEqual(download_state.get_status(), DLSTATUS_DOWNLOADING)
+        self.assertIsNone(download_state.get_error())
+
+        mock_transferred = MockObject()
+        mock_transferred.upTotal = 5
+        mock_transferred.downTotal = 6
+        mock_transferred.numConInitiated = 10
+        mock_transferred.numSeeds = 11
+        mock_transferred.numPeers = 12
+
+        stats = {'up': 123, 'down': 1234, 'stats': mock_transferred, 'time': 42, 'vod_prebuf_frac_consec': 43,
+                 'vod_prebuf_frac': 44, 'vod': True, 'tracker_status': {'a': 'b'}}
+        self.assertEqual(download_state.get_current_speed(UPLOAD), 0)
+        self.assertEqual(download_state.get_total_transferred(UPLOAD), 0)
+        self.assertEqual(download_state.get_num_peers(), 0)
+        self.assertEqual(download_state.get_num_nonseeds(), 0)
+        self.assertEqual(download_state.get_num_seeds_peers(), (0, 0))
+        self.assertEqual(download_state.get_vod_prebuffering_progress_consec(), 0)
+        self.assertEqual(download_state.get_vod_prebuffering_progress(), 0)
+        download_state.status = DLSTATUS_STOPPED
```

Comment:
There some logic that relies on `DLSTATUS_SEEDING` in the init method. I've added some small tests to also test this init method.